### DISCUSSION TITLE
3.2.2 On Input - Select for the number of pages to be displayed below a table

### DISCUSSION THREAD

**Discussion by Sommerli2** [w3c/wcag#3372] [Q&A]
We have discussed the following internally and need therefore some clarification: It’s all about a selection list for setting the displayed data records in a table, which is located below a table in combination with a pagination. Because the select responds to the "onChange" event - which immediately changes the number of rows in the table - we're not sure if this is an unexpected change of context or even just a change of content ("change of content"). Key word: change of viewport.

Does this quite common constellation/pattern conform to the success criterion 3.2.2 On Input or does the success criterion fails at this point?

If this constellation causes failure, how could possible solutions look like. An additional button as described in H84 would not be acceptable from a design point of view. Are there other alternatives?

**Comment by patrickhlauke** [comment]
Change of context is generally when the user's focus gets yanked around (an unexpected new page load, focus being shunted into another part of the current page), OR when the page dynamically changes to such a degree that the user is now effectively on a new/different page (think dynamic/AJAXy type pages, or single-page web apps, where you technically never left the page, but everything on it has changed dramatically).

In this case, as long as focus isn't changed as well (user is still on the select), just updating the following table is, to me, not a change of context.

As an aside, if there's anything like a status along the lines of "Showing 1-350 or 500" that gets updated, make sure this change is announced with a live region to satisfy 4.1.3 Status Messages.
